sql: mark implicit partition columns in information_schema.pg_indexes

otan opened the pull request:
Refs: #63203 

Release note (bug fix): Fix a case where implicitly partitioned columns
(e.g. from REGIONAL BY ROW tables and hash sharded indexes) previously
showed as implicit = false when using SHOW INDEXES /
information_schema.pg_indexes.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/63183)
<!-- Reviewable:end -->

ajstorm reviewed (approved):
> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63183)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajstorm

otan commented:
bors r-

otan committed:
sql: mark implicit partition columns in information_schema.pg_indexes

Release note (bug fix): Fix a case where implicitly partitioned columns
(e.g. from REGIONAL BY ROW tables and hash sharded indexes) previously
showed as implicit = false when using SHOW INDEXES /
information_schema.pg_indexes.

otan commented:
bors r=ajstorm

will backport for 21.1.1
